src/Log/LogServiceProvider.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Give\Log;
+
+use Give\Log\ValueObjects\LogCategory;
+use Give\ServiceProviders\ServiceProvider;
+use Give\Framework\Migrations\MigrationsRegister;
+use Give\Log\Migrations\CreateNewLogTables;
+use Give\Log\Migrations\MigrateExistingLogs;
+use Give\Log\Migrations\DeleteOldLogTables;
+
+/**
+ * Class LogServiceProvider
+ * @package Give\Log
+ *
+ * @since 2.9.7
+ */
+class LogServiceProvider implements ServiceProvider {
+	/**
+	 * @inheritdoc
+	 */
+	public function register() {
+		global $wpdb;
+
+		$wpdb->give_log = "{$wpdb->prefix}give_log";
+
+		give()->singleton( LogRepository::class );
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function boot() {
+		give( MigrationsRegister::class )->addMigrations(
+			[
+				CreateNewLogTables::class,
+			//              MigrateExistingLogs::class,
+			//              DeleteOldLogTables::class,
+			]
+		);
+
+		// Test it
+		if ( CreateNewLogTables::check() ) {
+			$this->testLogs();
+		}
+
```

Comment:
Remember to remove this before merging (don't resolve this comment until they're removed).

Reply:
Resolved
0265df80